addons/partner_autocomplete/models/res_partner.py

```diff
@@ -198,37 +198,40 @@ def check_gst_in(self, vat):
             return any(re.match(rx, vat) for rx in all_gstin_re)
         return False
 
-    def _is_synchable(self):
-        already_synched = self.env['res.partner.autocomplete.sync'].search([('partner_id', '=', self.id), ('synched', '=', True)])
-        return self.is_company and self.partner_gid and not already_synched
-
-    def _update_autocomplete_data(self, vat):
-        self.ensure_one()
-        if vat and self._is_synchable() and self._is_vat_syncable(vat):
-            self.env['res.partner.autocomplete.sync'].sudo().add_to_queue(self.id)
+    def _filter_synchable(self):
+        companies = self.filtered('is_company').filtered('partner_gid')
+        already_synched = set(self.env['res.partner.autocomplete.sync'].search([
+            ('partner_id', 'in', companies.ids), ('synched', '=', True)]
+        ).partner_id.ids)
+        return companies.filtered(lambda p: p.id not in already_synched)
+
+    def _update_autocomplete_data(self):
+        for partner in self._filter_synchable():
+            vat = partner.vat
+            if vat and partner._is_vat_syncable(vat):
+                self.env['res.partner.autocomplete.sync'].sudo().add_to_queue(partner.id)
 
     @api.model_create_multi
     def create(self, vals_list):
-        partners = super(ResPartner, self).create(vals_list)
-        if len(vals_list) == 1:
-            partners._update_autocomplete_data(vals_list[0].get('vat', False))
-            if partners.additional_info:
-                template_values = json.loads(partners.additional_info)
+        partners = super().create(vals_list)
+        partners._update_autocomplete_data()
+        for partner in partners:
+            if partner.additional_info:
+                template_values = json.loads(partner.additional_info)
```

Comment:
I think the `len(...) == 1` checks were done on purpose to only update the partner when it is created/updated through the form view.

Reply:
Well, you can update one record from outside the view too. Maybe skip that when `import_file` is set? What do you think?

Reply:
Yes, I know, it's sketchy.
Anyways, this part of the code will be removed soon, once this [task](https://www.odoo.com/odoo/project/967/tasks/4416928) is done, so I wouldn't bother too much with it.

Reply:
Ok, so this is a temporary improvement and will be replaced in the near future.